zcash_primitives/CHANGELOG.md

```diff
@@ -95,19 +95,35 @@ and this library adheres to Rust's notion of
     newtypes.
   - `address::PaymentAddress::create_note` now takes its `value` argument as a
     `NoteValue` instead of as a bare `u64`.
+  - `builder::SaplingBuilder` no longer has a `P: consensus::Parameters` type
+    parameter.
+  - `builder::SaplingBuilder::new` now takes a `canopy_active` boolean instead
+    of a `P: consensus::Parameters` argument and a target height.
   - `builder::SaplingBuilder::add_spend` now takes `extsk` by reference.
   - `builder::SaplingBuilder::build` no longer takes a prover, proving context,
-    or progress notifier. Instead, it has `SpendProver, OutputProver` generic
-    parameters and returns `(UnauthorizedBundle, SaplingMetadata)`. The caller
-    can then use `Bundle::<InProgress<Unproven, _>>::create_proofs` to create
-    spend and output proofs for the bundle.
+    progress notifier, or target height. Instead, it has `SpendProver, OutputProver`
+    generic parameters and returns `(UnauthorizedBundle, SaplingMetadata)`. The
+    caller can then use `Bundle::<InProgress<Unproven, _>>::create_proofs` to
+    create spend and output proofs for the bundle.
   - `builder::Error` has new error variants:
     - `Error::DuplicateSignature`
     - `Error::InvalidExternalSignature`
     - `Error::MissingSignatures`
   - `bundle::MapAuth` trait methods now take `&mut self` instead of `&self`.
   - `circuit::ValueCommitmentOpening::value` is now represented as a `NoteValue`
     instead of as a bare `u64`.
+  - `note_encryption`:
+    - `SaplingDomain` no longer has a `P: consensus::Parameters` type parameter.
+    - The following methods now take a Canopy activation height instead of a
+      `P: consensus::Parameters` argument:
+      - `SaplingDomain::for_height`
+      - `plaintext_version_is_valid`
+      - `try_sapling_note_decryption`
+      - `try_sapling_compact_note_decryption`
+      - `try_sapling_output_recovery_with_ock`
+      - `try_sapling_output_recovery`
+  - `util::generate_random_rseed` now takes a `canopy_active` boolean instead of
+    a `P: consensus::Parameters` argument and a height.
```

Comment:
Hmmm. While I understand the rationale for this change, you now have a method that takes two `BlockHeight` arguments with very different semantics, and it would be really easy to get them confused. I would prefer if we had an enum that distinguished between the `PreZip212` and `PostZip212` states.